src/modules/hip/kernel/magnitude.hpp

```diff
@@ -0,0 +1,276 @@
+#include <hip/hip_runtime.h>
+#include "rpp_hip_common.hpp"
+
+__device__ void magnitude_hip_compute(uchar *srcPtr, d_float8 *src1_f8, d_float8 *src2_f8, d_float8 *dst_f8)
+{
+    dst_f8->f1[0] = sqrtf(fmaf(src1_f8->f1[0], src1_f8->f1[0], src2_f8->f1[0] * src2_f8->f1[0]));
+    dst_f8->f1[1] = sqrtf(fmaf(src1_f8->f1[1], src1_f8->f1[1], src2_f8->f1[1] * src2_f8->f1[1]));
+    dst_f8->f1[2] = sqrtf(fmaf(src1_f8->f1[2], src1_f8->f1[2], src2_f8->f1[2] * src2_f8->f1[2]));
+    dst_f8->f1[3] = sqrtf(fmaf(src1_f8->f1[3], src1_f8->f1[3], src2_f8->f1[3] * src2_f8->f1[3]));
+    dst_f8->f1[4] = sqrtf(fmaf(src1_f8->f1[4], src1_f8->f1[4], src2_f8->f1[4] * src2_f8->f1[4]));
+    dst_f8->f1[5] = sqrtf(fmaf(src1_f8->f1[5], src1_f8->f1[5], src2_f8->f1[5] * src2_f8->f1[5]));
+    dst_f8->f1[6] = sqrtf(fmaf(src1_f8->f1[6], src1_f8->f1[6], src2_f8->f1[6] * src2_f8->f1[6]));
+    dst_f8->f1[7] = sqrtf(fmaf(src1_f8->f1[7], src1_f8->f1[7], src2_f8->f1[7] * src2_f8->f1[7]));
+}
+
+__device__ void magnitude_hip_compute(float *srcPtr, d_float8 *src1_f8, d_float8 *src2_f8, d_float8 *dst_f8)
+{
+    dst_f8->f1[0] = sqrtf(fmaf(src1_f8->f1[0], src1_f8->f1[0], src2_f8->f1[0] * src2_f8->f1[0]));
+    dst_f8->f1[1] = sqrtf(fmaf(src1_f8->f1[1], src1_f8->f1[1], src2_f8->f1[1] * src2_f8->f1[1]));
```

Comment:
All the src2_f8 squaring could be f4 vectorized

Reply:
Yes changed this, I was checcking perf yesterday and its better using multiply and add than fmaf so modified this accordingly.